engines/instance_verification/lib/instance_verification/engine.rb

```diff
@@ -15,16 +15,15 @@ def self.update_cache(cache_entry, mode, registry: false)
     )
   end
 
-  def self.build_cache_entry(remote_ip, system_login, encoded_reg_code, mode, product)
+  def self.build_cache_entry(remote_ip, system_login, encoded_reg_code, mode, product, product_class) # rubocop:disable Metrics/ParameterLists
```

Comment:
Can we move `product_class = product.product_class` into this method, to not make the parameter list even longer?

Reply:
no, because the `registry` engine uses `product_class` and does not have a product as it is now, see the [call](https://github.com/SUSE/rmt/blob/master/engines/registry/app/models/access_scope.rb#L100)

could we get the product with a DB query ? we are trying to reduce DB queries too

Reply:
Sorry, I do not quite understand. The parameter list already includes the product object and product_class is a member of that object. Why pass the object and a member of the object when inside the method we could access the member explicitly. Or is the `product` argument misnamed?

Reply:
we could change the logic in the registry method
```python
allowed_non_free_product_classes = allowed_product_classes.map { |s| s unless Product.find_by(product_class: s, product_type: 'extension').free? }.compact
```
to use the whole Product instead of its product class

Reply:
instead of using the extra param and not touch the registry allow path code,refactored that part and use product only